static/js/apps/base/components/HeaderBar.tsx

```diff
@@ -0,0 +1,120 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement, useMemo } from "react";
+
+import { HeaderMenu } from "../../../shared/types/base";
+import { Labels, Routes } from "../../../shared/types/general";
+import { resolveHref } from "../utilities/utilities";
+
+interface HeaderBarProps {
+  name: string;
+  logoPath: string;
+  menu: HeaderMenu[];
+  labels: Labels;
+  routes: Routes;
+}
+
+const HeaderBar = ({
+  name,
+  logoPath,
+  menu,
+  labels,
+  routes,
+}: HeaderBarProps): ReactElement => {
+  const visibleMenu = useMemo(() => {
+    return menu.map((menuItem) => ({
+      ...menuItem,
+      subMenu: menuItem.subMenu.filter((subMenuItem) => !subMenuItem.hide),
+    }));
+  }, [menu]);
+
+  return (
+    <header id="main-header">
+      <nav className="navbar navbar-light navbar-expand-lg col" id="main-nav">
+        <div className="container-fluid">
+          <div className="navbar-brand">
+            {logoPath && (
+              <div id="main-header-logo">
+                <a
+                  href={routes["static.homepage"]}
+                  aria-label={labels["Back to homepage"]}
+                >
+                  <img src={logoPath} alt={`${name} logo`} />
+                </a>
+              </div>
+            )}
+            <a href={routes["static.homepage"]}>{name}</a>
+          </div>
+          <button
+            className="navbar-toggler"
+            type="button"
+            data-toggle="collapse"
+            data-target="#dc-main-nav"
+            aria-label={labels["Show site navigation"]}
+          >
+            <span className="navbar-toggler-icon"></span>
+          </button>
+          <div
+            className="collapse navbar-collapse justify-content-md-end"
+            id="dc-main-nav"
+          >
+            <ul className="navbar-nav float-md-right">
+              {visibleMenu.map((menuItem, index) => (
+                <li key={menuItem.label} className="nav-item dropdown">
+                  <a
+                    className="nav-link dropdown-toggle"
+                    href="#"
+                    id="nav-explore-dropdown"
+                    role="button"
+                    data-toggle="dropdown"
+                    aria-haspopup="true"
+                    aria-expanded="false"
+                    aria-label={labels[menuItem.ariaLabel]}
+                  >
+                    {labels[menuItem.label]}
+                  </a>
+                  {menuItem.subMenu.length > 0 && (
+                    <div
+                      className={`dropdown-menu ${
+                        index === visibleMenu.length - 1
+                          ? "dropdown-menu-right"
+                          : ""
+                      }`}
+                      aria-labelledby="nav-explore-dropdown"
```

Comment:
Each of the dropdown menus generated by the loop uses the same id ("nav-explore-dropdown") in the `aria-labelledby` attribute, whereas it should use a unique `id`. Maybe by appending or prepending the loop index would do the trick.